game/midifile.cc

```diff
@@ -34,40 +36,40 @@ class MidiStream {
 	}
 
 	/// read bytes
-	std::string read_bytes(size_t bytes);
+	std::string read_bytes(std::streamoff bytes);
 
 	class Riff {
 	  public:
 		MidiStream& ms;
 		std::string name;
-		size_t pos;
-		size_t size;
-		size_t offset;
+		std::ptrdiff_t pos;
+		std::ptrdiff_t size;
+		std::ptrdiff_t offset;
 		Riff(MidiStream& ms);
 		~Riff();
 		bool has_more_data() const { return offset < size; }
-		uint8_t read_uint8() { consume(1); return ms.f.get(); }
-		uint16_t read_uint16() { consume(2); return ms.read_uint16(); }
-		uint32_t read_uint32() { consume(4); return ms.read_uint32(); }
-		uint32_t read_varlen();
+		std::uint8_t read_uint8() { consume(1); return static_cast<std::uint8_t>(ms.f.get()); }
+		std::uint16_t read_uint16() { consume(2); return ms.read_uint16(); }
+		std::uint32_t read_uint32() { consume(4); return ms.read_uint32(); }
+		std::uint32_t read_varlen();
 		template <typename T> T read(T& value) {
 			consume(sizeof(T));
 			value = 0;
-			for (int i = sizeof(T) - 1; i >= 0; --i) value |= ms.f.get() << (8 * i);
+			for (int i = sizeof(T) - 1; i >= 0; --i) value |= static_cast<T>(ms.f.get() << (8 * i));
 			return value;
 		}
-		std::string read_bytes(size_t size) { consume(size); return ms.read_bytes(size); }
-		void ignore(size_t size) { consume(size); ms.f.ignore(size); }
-		void seek_back(size_t offset = 1);
+		std::string read_bytes(std::uint32_t size) { consume(size); return ms.read_bytes(size); }
+		void ignore(std::streamoff size) { consume(size); ms.f.ignore(size); }
+		void seek_back(std::streamoff offset = 1);
 	  private:
-		void consume(size_t bytes);
+		void consume(std::streamoff bytes);
 	};
 
 private:
 
 	std::stringstream f;
-	uint16_t read_uint16() { return f.get() << 8 | f.get(); }
-	uint32_t read_uint32() { return f.get() << 24 | f.get() << 16 | f.get() << 8 | f.get(); }
+	std::uint16_t read_uint16() { return static_cast<std::uint16_t>(f.get() << 8 | f.get()); }
+	std::uint32_t read_uint32() { return static_cast<std::uint32_t>(f.get() << 24 | f.get() << 16 | f.get() << 8 | f.get()); }
```

Comment:
I'd suggest 
uint32_t v = f.get(); and doing bitshift on 32bits. keeping the cast after bit shifts is making it more strange.